src/components/modals/progressModal.tsx

```diff
@@ -0,0 +1,33 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import Modal from "@/components/common/containers/modal";
+import Spinner from "../common/containers/spinner";
+import BoatSvg from "../../assets/svgs/BoatSvg";
+
+export default function ProcessingModal({
+  isOpen,
+}: {
+  isOpen: boolean;
+}) {
```

Comment:
I had imagine we put all the logic here to check what needs to be displayed and open / close the other modals